src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/update_device_list_request.cc

```diff
@@ -53,51 +53,27 @@ UpdateDeviceListRequest::UpdateDeviceListRequest(
                    application_manager,
                    rpc_service,
                    hmi_capabilities,
-                   policy_handle)
-    , EventObserver(application_manager_.event_dispatcher()) {}
+                   policy_handle) {}
 
 UpdateDeviceListRequest::~UpdateDeviceListRequest() {}
 
 void UpdateDeviceListRequest::Run() {
   SDL_LOG_AUTO_TRACE();
-  sync_primitives::AutoLock auto_lock(wait_hmi_lock);
   // Fix problem with SDL and HMI HTML. This problem is not actual for HMI PASA.
   // Flag conditional compilation for specific customer is used in order to
   // exclude
   // hit code to RTC
-  if (true == application_manager_.get_settings().launch_hmi()) {
-    if (!application_manager_.IsHMICooperating()) {
-      SDL_LOG_INFO("Wait for HMI Cooperation");
-      subscribe_on_event(hmi_apis::FunctionID::BasicCommunication_OnReady);
-      termination_condition_.Wait(auto_lock);
-      SDL_LOG_DEBUG("HMI Cooperation OK");
+  if (application_manager_.get_settings().launch_hmi()) {
```

Comment:
As I remember, we will always have this condition `false` (`application_manager_.get_settings().launch_hmi()`), because `launch_hmi` value by default is assigned to `false` in `smartdeviceLink.ini` file, and now we cannot have situation when `launch_hmi` may be `true`. So this check may be redundant and now in fact we have never wait for HMI cooperation in this request.

Reply:
@ychernysheva well, technically you can set `LaunchHMI` to true in ini file. It might have no sense, but it is possible and unless it is possible, this code can't be considered as a dead code so it should be maintained too. By that reason, I prefer to keep this branch for now. Removal of entire parameter is a topic of another proposal

Reply:
If I understand correctly, now actually we don't need to wait anything here, and this code block is left only for backward compatibility reason?

Reply:
@ychernysheva we need to wait for HMI readiness here. It was done using `on_event`, now I just replaced that with another waiting mechanism. You are right that code block itself is a backward compatibility code.